src/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs

```diff
@@ -158,37 +166,28 @@ public static class ClientOperations
                 ("GET",
                 async ctx =>
                 {
-                    using (var req = new HttpRequestMessage(HttpMethod.Get, "/"))
-                    using (HttpResponseMessage m = await ctx.SendAsync(req))
-                    {
-                        ValidateHttpVersion(m, ctx.HttpVersion);
-                        ValidateStatusCode(m);
-                        string content = await m.Content.ReadAsStringAsync();
-                        if (content.Length != ctx.MaxContentLength)
-                        {
-                            throw new Exception($"Expected content length {ctx.MaxContentLength} but got {content.Length}");
-                        }
-
-                        if (!ServerContentUtils.IsValidServerContent(content))
-                        {
-                            throw new Exception("Unexpected response content");
-                        }
-                    }
+                    using var req = new HttpRequestMessage(HttpMethod.Get, "/");
+                    int expectedLength = ctx.SetExpectedResponseContentLengthHeader(req.Headers);
+                    using HttpResponseMessage m = await ctx.SendAsync(req);
+                    
+                    ValidateHttpVersion(m, ctx.HttpVersion);
+                    ValidateStatusCode(m);
+                    ValidateServerContent(await m.Content.ReadAsStringAsync(), expectedLength);
                 }),
 
                 ("GET Partial",
                 async ctx =>
                 {
-                    using (var req = new HttpRequestMessage(HttpMethod.Get, "/slow"))
-                    using (HttpResponseMessage m = await ctx.SendAsync(req, HttpCompletionOption.ResponseHeadersRead))
-                    {
-                        ValidateHttpVersion(m, ctx.HttpVersion);
-                        ValidateStatusCode(m);
+                    using var req = new HttpRequestMessage(HttpMethod.Get, "/slow");
+                    int expectedLength = ctx.SetExpectedResponseContentLengthHeader(req.Headers, minLength: 2);
+                    using HttpResponseMessage m = await ctx.SendAsync(req, HttpCompletionOption.ResponseHeadersRead);
 
-                        using (Stream s = await m.Content.ReadAsStreamAsync())
-                        {
-                            s.ReadByte(); // read single byte from response and throw the rest away
-                        }
+                    ValidateHttpVersion(m, ctx.HttpVersion);
+                    ValidateStatusCode(m);
+
+                    using (Stream s = await m.Content.ReadAsStreamAsync())
+                    {
+                        s.ReadByte(); // read single byte from response and throw the rest away
```

Comment:
Is this intended to be sync?

Reply:
Probably not. I'm assuming @stephentoub wrote it like this for brevity.

Reply:
Right, I didn't really care in this case whether it was sync or async, I just wanted simple. Feel free to change it if you please, or leave it, either way :)